### PR TITLE
add junos_pmtud module

### PR DESCRIPTION
**Pull request by mkomon**
Add junos_pmtud module to perform path MTU discovery on Junos devices. 

Finds path MTU to destination IP, with defaults it finds MTU in range 989-1500 B in exactly 9 steps (always log2(max_range) steps). Tolerates any packet loss lower than 100 pct.

### CODE REVIEW

**Comment by stacywsmith** [comment]
@mkomon I like the idea of this module and will review the code today. On a related note...
Would you please consider submitting an additional pull request to add the `routing_instance` argument to `junos_ping`? That would also be helpful.

File: library/junos_pmtud [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        changed=False,
+        mtu=0,
+        dest_ip=m_args['dest_ip']
+    )
+
+    # Prepare parameters
+    ping_params = dict(
+            host=m_args['dest_ip'],
+            count='3',
+            do_not_fragment=True,
```
Comment (stacywsmith) [inline library/junos_pmtud]: As it currently stands, a user could enter an IPv6 address for `dest-ip` (or a hostname which resolves to an IPv6 address) and the module will return a value that is 20 bytes less than the actual IPv6 MTU because it assumes IPv4 sized header.

For now, we can force IPv4 only and avoid an incorrect answer by adding `inet=True` to `ping_params`

In the long run, additional arguments and code could be added for other non-IPv4 address families.

File: library/junos_pmtud [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        step = int(m_args['max_range'])
+
+        while True:
+            step = step / 2 if step >= 2 else 0
+            ping_params['size'] = str(test_size)
+            rpc_reply = dev.rpc.ping(**ping_params)
+            loss = int(rpc_reply.find('probe-results-summary').findtext("packet-loss").strip())
+            if loss < 100 and test_size == int(m_args['max_size']):
+                # ping success with max test_size, save and break
+                results["mtu"] = test_size
```
Comment (stacywsmith) [inline library/junos_pmtud]: See comment on line 187.

File: library/junos_pmtud [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+                break
+            elif loss < 100:
+                # ping success, increase test_size
+                results["mtu"] = test_size
```
Comment (stacywsmith) [inline library/junos_pmtud]: See comment on line 187.

File: library/junos_pmtud [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    except Exception as err:
+        results['failed'] = True
+        results['msg'] = "unable to execute ping due to:{0}".format(err.message)
+        dev.close()
+        raise err
+
+    if not results["mtu"]:
+        module.fail_json(msg='MTU too low, increase max_range.', **results)
+    else:
+        results["mtu"] += 28    # adjust for IP and ICMP headers
```
Comment (stacywsmith) [inline library/junos_pmtud]: See comment on line 187.

File: library/junos_pmtud (hunk 1 above)
Comment (mkomon) [reply library/junos_pmtud]: Admittedly, I had only IPv4 in mind when writing this. At the moment I don't have environment with IPv6 running on JUNOS so I agree to limit the module to IPv4 now.

File: library/junos_pmtud [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    ## Change default Timeout
+    timeout = int(m_args['timeout'])
+    if timeout > 0:
+        dev.timeout = timeout
+
+    try:
+        # Check if ICMP passes
+        ping_params['size'] = str(64)
+        rpc_reply = dev.rpc.ping(**ping_params)
+        loss = int(rpc_reply.find('probe-results-summary').findtext("packet-loss").strip())
```
[the remark this answers is not in the document]
Comment (mkomon) [reply library/junos_pmtud]: Thanks for the hint! Fixed.

**Comment by stacywsmith** [comment]
@mkomon It looks like you've address all of my feedback except for changing the result key from `mtu` to `inet_mtu`. Let me know if you disagree or have an alternate suggestion.

File: library/junos_pmtud (hunk 3 above)
Comment (mkomon) [reply library/junos_pmtud]: At first I didn't get this one. Sure, inet_mtu makes sense, I've pushed a new commit.
